### PR TITLE
r.out.gdal: raster attribute table export enhancement

### PR DESCRIPTION
**Pull request by metzm**
This PR solves an old TODO by writing both labels and colors to a raster attribute table (RAT) with `r.out.gdal -t`.

Newer QGIS versions can now also read raster attribute tables and will use the labels and/or colors listed in a RAT.

Test with
```
g.region -p rast=landcover_1m
r.out.gdal in=landcover_1m out=landcover_1m.tif format=GTiff createopt="COMPRESS=DEFLATE" -t
gdalino landcover_1m.tif
```
in the NC sample dataset.

### CODE REVIEW

**Review by mmacata — APPROVED** [review]
Very cool, thanks!

**Comment by neteler** [comment]
Works smoothly! I have tested it with a different dataset, both in GRASS GIS and QGIS:

image

**Review by neteler — APPROVED** [review]

File: raster/r.out.gdal/attr.c [hunk 1]
```diff
@@ -47,12 +49,149 @@ int export_attr(GDALDatasetH hMEMDS, int band, const char *name,
...
+
+        /* merge categories and color rules:
+         * go through categories and fetch corresponding color
+         * write out value, label, red, green, blue
+         * or the minmax variant
+         */
+
+        if (maptype == CELL_TYPE) {
+            for (i = 0; i < cats.ncats; i++) {
+                label = Rast_get_ith_c_cat(&cats, i, &CellMin, &CellMax);
```
Comment (nilason) [inline raster/r.out.gdal/attr.c:64]: This looks very good to me, and a great addition!

I have only minor suggestions: (1) `label` which is set on L64 and L73 (and L199 and L208) are never used: the return value may just as well be ignored on these lines; (2) `use_minmax` would be better represented by a bool.

Comment (metzm) [reply raster/r.out.gdal/attr.c:64]: (1) done and (2) done. OK?

Comment (metzm) [reply raster/r.out.gdal/attr.c:64]: I understand, but with `(void)Rast_get_ith_c_cat()` you just trick the format checker to not complain about unused return codes. Therefore I think my proposed solution where the code now also checks if the call to `Rast_get_ith_c_cat()` was successful (there is someting in label) is the formally correct solution.

Comment (nilason) [reply raster/r.out.gdal/attr.c:64]: (This came up with `scan-build`, as "unused assignment" or something similar. Occasionally, I run it on new code or substantial updates).

I didn't check for possible erroring of `Rast_get_ith_c_cat()`, so with your update there's actually a code improvement (not only for better "code reading").

**Review by nilason — APPROVED** [review]
Thanks!

**Pull request merged**